packages/core/src/components/Form.js

```diff
@@ -39,6 +39,12 @@ const formDisabledStateHasChanged = (
 ) =>
   nextProps.disabled !== undefined && nextProps.disabled !== prevState.disabled;
 
+const formTouchedStateHasChanged = (
+  nextProps: FormComponentProps,
+  prevState: FormComponentState
+) =>
+  nextProps.touched !== prevState.touched;
```

Comment:
If we change this to use `showValidationBeforeTouched` (see comment below) then I think this would be better - but the name of the function might want to change (e.g. `formTouchedBehaviourHasChanged` maybe?)